include/dem/force_chains_visualization.h

```diff
@@ -136,30 +129,71 @@ class ParticlesForceChains
+   * contact forces.
+   * @param[in] ghost_adjacent_particles Container of the contact pair
+   * candidates information for calculation of the local-ghost particle-particle
+   * contact forces.
    */
   void
   write_force_chains(const DEMSolverParameters<dim> &dem_parameters,
                      PVDHandler                     &pvd_handler,
                      const MPI_Comm                  mpi_communicator,
                      const std::string               folder,
                      const unsigned int              iter,
-                     const double                    time) override;
+                     const double                    time,
+                     typename dem_data_structures<dim>::adjacent_particle_pairs
+                       &local_adjacent_particles,
+                     typename dem_data_structures<dim>::adjacent_particle_pairs
+                       &ghost_adjacent_particles) override;
 
 private:
+  /**
+   * @brief Calculate normal forces between all touching particles with
+   * ParticleParticleContactForce class' methods. Stock normal forces and
+   * particles position in vectors.
+   *
+   * @param[in] local_adjacent_particles Container of the contact pair
+   * candidates information for calculation of the local particle-particle
+   * contact forces.
+   * @param[in] ghost_adjacent_particles Container of the contact pair
+   * candidates information for calculation of the local-ghost particle-particle
+   * contact forces.
+   * @param[out] vertices the vector of positions of touching particles.
+   * @param[out] normal_forces_vector the vector of normal forces between each
+   * touching particles.
+   */
+  void
+  calculate_force_chains(
+    typename dem_data_structures<dim>::adjacent_particle_pairs
+      &local_adjacent_particles,
+    typename dem_data_structures<dim>::adjacent_particle_pairs
+                          &ghost_adjacent_particles,
+    std::vector<Point<3>> &vertices,
+    std::vector<double>   &normal_forces_vector) override;
+
   /**
    * @brief Execute the contact calculation step for the particle-particle
    * contact only for local-local and local-ghost contacts with no periodicity.
    * This is a simplified version of the contact calculation of the
    * particle-particle contact forces class, without the other contact types and
    * the update of the particles forces, torques and tangential overlap.
    *
+   * @tparam force_model The particle-particle contact force model.
    * @param[in] adjacent_particles_list Container of the adjacent particles of a
    * particles.
+   * @param[out] vertices the vector of positions of touching particles.
+   * @param[out] normal_forces_vector the vector of normal forces between each
+   * touching particles.
    */
+  template <ContactType contact_type>
   inline void
   execute_contact_calculation(
     typename DEM::dem_data_structures<dim>::particle_contact_info
```

Comment:
This could be const

Reply:
No it cannot. It has something to do with the particle_particle_contact_force.h file. 

There's an other function called `execute_contact_calculation` with the same input parameters. In that function, we clear the tangential overlap, thus it is not const. 

I'm not sure if the ParticlesForceChains class needs to inherit from ParticleParticleContactForce, but this is way it cannot be const.

Reply:
Perfect